src/main/resources/assets/js/app/report/ReportProgressList.ts

```diff
@@ -2,36 +2,73 @@ import ListBox = api.ui.selector.list.ListBox;
 import DateHelper = api.util.DateHelper;
+import TaskEvent = api.task.TaskEvent;
+import TaskEventType = api.task.TaskEventType;
 import {Report} from './Report';
-import {ReportWebSocket} from './ReportWebSocket';
 import {ListReportsRequest} from '../../api/graphql/report/ListReportsRequest';
 import {DeleteReportsRequest} from '../../api/graphql/report/DeleteReportsRequest';
+import {GetReportRequest} from '../../api/graphql/report/GetReportRequest';
 
 export class ReportProgressList
     extends ListBox<Report> {
 
     constructor(principalKey: PrincipalKey, repositoryIds?: string[]) {
         super('report-progress-list');
-        const socket = ReportWebSocket.getInstance();
 
         this.loadExistingReports(principalKey, repositoryIds);
+        this.handlePermReportsTaskEvents();
+    }
 
-        socket.onReportProgress((report: Report, progress: number) => {
-            let view = <ReportProgressItem> this.getItemView(report);
-            if (!view) {
-                this.addItem(report);   // report has not yet been added but the progress has already started
-                view = <ReportProgressItem> this.getItemView(report);
-            }
+    private handlePermReportsTaskEvents() {
+        const handler = this.handleTaskEvent.bind(this);
 
-            if (progress < 100) {
-                view.setProgress(progress);
-            } else {
-                view.setFinished(report.getFinished());
-                view.setReportReady(true);
-            }
+        TaskEvent.on(handler);
+
+        this.onRemoved(() => {
+            TaskEvent.un(handler);
         });
+    }
+
+    private handleTaskEvent(event: TaskEvent) {
+        if (event.getTaskInfo().getDescription().indexOf('Report task') < 0) {
+            return;
+        }
+
+        const view: ReportProgressItem = this.getItemViewByTaskEvent(event);
+
+        if (!view) {
+            return;
+        }
+
+        if (event.getEventType() === TaskEventType.UPDATED) {
+            const taskProgress: api.task.TaskProgress = event.getTaskInfo().getProgress();
+            const progress: number = taskProgress.getCurrent() * 100 / taskProgress.getTotal();
+            view.setProgress(progress);
+
+            return;
+        }
 
-        socket.onReportDownload((report: Report) => this.deleteReport(report));
+        if (event.getEventType() === TaskEventType.FINISHED) {
+            view.setReportReady(true);
+
+            new GetReportRequest(view.getItem().getId()).sendAndParse().then((report: Report) => {
+                view.setFinished(report.getFinished());
+            });
+
+            api.notify.NotifyManager.get().showSuccess(i18n('notify.report.finished', view.getItem().getPrincipalDisplayName()));
+
+            return;
+        }
+    }
+
+    private getItemViewByTaskEvent(event: TaskEvent): ReportProgressItem {
+        const items = this.getItems().filter((item: Report) => item.getTaskId() === event.getTaskInfo().getId().toString());
+
```

Comment:
It's not a biggie - especially considering the number of ``items`` - but ``event.getTaskInfo().getId().toString()`` will always be the same, so it's more efficient to get its value before the loop and use it inside the loop.

Reply:
True, missed that , updating